datadog_checks_dev/datadog_checks/dev/ssh_tunnel.py

```diff
@@ -0,0 +1,88 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import absolute_import
+
+import os
+import socket
+from contextlib import closing, contextmanager
+
+from six import PY3
+
+from .env import environment_run
+from .structures import LazyFunction, TempDir
+from .subprocess import run_command
+
+if PY3:
+    import subprocess
+else:
+    import subprocess32 as subprocess
+
+
+def find_free_port():
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(('', 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return s.getsockname()[1]
+
+
+def get_ip():
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_DGRAM)) as s:
+        # doesn't even have to be reachable
+        s.connect(('10.255.255.255', 1))
+        return s.getsockname()[0]
+
+
+@contextmanager
+def socks_proxy(host, user, private_key):
+    """Open a SSH connection with a SOCKS proxy."""
+    set_up = SocksProxyUp(host, user, private_key)
+    tear_down = SocksProxyDown()
+    conditions = []
```

Comment:
Is there a reason to specify an empty `conditions`?

Reply:
Nope removed.